src/mds/CInode.cc

```diff
@@ -5165,6 +5165,11 @@ void CInode::dump(Formatter *f, int flags) const
     }
     f->close_section();
   }
+
+  auto realm = find_snaprealm();
+  inodeno_t subvol_ino = realm->get_subvolume_ino();
+  bool is_subvol = (subvol_ino && subvol_ino == ino());
```

Comment:
`subvol_ino && (subvol_ino == ino())` be better right

Reply:
The equality operator has higher precedence than the logical AND operator as per [C++ Operator Precedence Rules](https://en.cppreference.com/w/cpp/language/operator_precedence).

Reply:
in that case do we really need the outer parenthesis? `bool is_subvol = subvol_ino && subvol_ino == ino();` would suffice